spec/lib/bundler/explain/parser_spec.rb

```diff
@@ -2,10 +2,26 @@
 
 describe Bundler::Explain::Parser do
   describe '#call' do
-    it 'process file to hash' do
-      file_path = "#{File.dirname(__FILE__)}/../../../examples/case1/Gemfile.lock"
-      result = Bundler::Explain::Parser.new(file_path).call
-      expect(result.dig('GEM', 'specs', 'actioncable (5.1.4)').keys).to match_array ['actionpack (= 5.1.4)', 'nio4r (~> 2.0)', 'websocket-driver (~> 0.6.1)']
+    let(:gemfile) { File.expand_path("#{File.dirname(__FILE__)}/../../../examples/case1/Gemfile") }
+    let(:gemfile_lock) { File.expand_path("#{File.dirname(__FILE__)}/../../../examples/case1/Gemfile.lock") }
+
+    subject { Bundler::Explain::Parser.new(gemfile: gemfile, gemfile_lock: gemfile_lock).call }
+
+    it 'processing success' do
+      expect(subject.success?).to eq true
+    end
+
+    it 'return dependencies from Gemfile.lock' do
+      expect(subject.dependencies.count).to eq 69
+      expect(subject.dependencies["activesupport"]).to match_array ["concurrent-ruby", "i18n", "minitest", "tzinfo"]
+      expect(subject.dependencies["actioncable"]).to match_array ["actionpack", "nio4r", "websocket-driver"]
+    end
+
+    it 'return dependencies from Gemfile' do
+      expect(subject.from_gemfile).to match_array %w(
```

Comment:
Again `describe '#from_gemfile'`. And, BTW, I don't think `from_gemfile` is a good name. Something like `direct`?